osu.Game.Rulesets.Taiko/Mods/TaikoModSingleTap.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Localisation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Taiko.Objects;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
+using osu.Game.Utils;
+using osu.Game.Rulesets.Taiko.UI;
+
+namespace osu.Game.Rulesets.Taiko.Mods
+{
+    public partial class TaikoModSingleTap : Mod, IApplicableToDrawableRuleset<TaikoHitObject>, IUpdatableByPlayfield
+    {
+        public override string Name => @"Single Tap";
+        public override string Acronym => @"SG";
+        public override LocalisableString Description => @"One key for dons, one key for kats.";
+
+        protected bool CheckValidNewAction(TaikoAction action)
+        {
+            if (action == TaikoAction.LeftCentre || action == TaikoAction.RightCentre)
+                return LastAcceptedDonAction == null || LastAcceptedDonAction == action;
+            if (action == TaikoAction.LeftRim || action == TaikoAction.RightRim)
+                return LastAcceptedKatAction == null || LastAcceptedKatAction == action;
+
+            return true;
+        }
+
+        public override double ScoreMultiplier => 1.0;
+        public override Type[] IncompatibleMods => new[] { typeof(ModAutoplay), typeof(ModRelax), typeof(TaikoModCinema) };
+        public override ModType Type => ModType.Conversion;
+
+        private DrawableTaikoRuleset ruleset = null!;
+
+        private TaikoPlayfield playfield { get; set; } = null!;
+
+        protected TaikoAction? LastAcceptedDonAction { get; private set; }
+        protected TaikoAction? LastAcceptedKatAction { get; private set; }
```

Comment:
- Please make `private` for now
- Can these be named `LastAccepted{Centre,Rim}Action`? I would rather avoid having to mentally parse "is centre don or kat again?" every time this code is read.